### PR TITLE
Implement php version dependent loose const comparison

### PR DESCRIPTION
**Pull request by staabm**
refs https://github.com/phpstan/phpstan-src/pull/2216#discussion_r1096637226

see https://www.php.net/manual/en/migration80.incompatible.php#migration80.incompatible.core.string-number-comparision

https://3v4l.org/pWAjV
https://3v4l.org/0VWtq

https://phpstan.org/r/b2748c38-b33e-4db7-a60f-4f5edd991270

### CODE REVIEW

File: tests/PHPStan/Analyser/looseConstComparisonPhp7.neon [hunk 1]
```diff
@@ -0,0 +1,2 @@
+parameters:
+	phpVersion: 70400 # PHP 7.4
```
Comment (staabm) [inline tests/PHPStan/Analyser/looseConstComparisonPhp7.neon:2]: when running the test-cases in isolation, outside of the phpstan repo they behave as expected. 

I somehow have the feeling that using `phpVersion` within the test-suite is kind of a unsupported feature in the current test infrastructure (I think we see some kind of cache problem)

Comment (ondrejmirtes) [reply tests/PHPStan/Analyser/looseConstComparisonPhp7.neon:2]: You need to change `new PhpVersion(PHP_VERSION_ID)` in PHPStanTestCase

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: tests/PHPStan/Analyser/looseConstComparisonPhp7.neon (hunk 1 above)
Comment (staabm) [reply tests/PHPStan/Analyser/looseConstComparisonPhp7.neon:2]: thank you.

**Comment by ondrejmirtes** [comment]
Thank you.

**Pull request merged**